src/Components/ProfileImportDialog.js

```diff
@@ -0,0 +1,267 @@
+// TODO Make these imports targeted, once set is finalized.
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import TextField from "@mui/material/TextField";
+
+import { useContext, useState } from "react";
+import { APIGetMalLists } from "./APICalls";
+import BreathingLogo from "./BreathingLogo";
+import { useSnackbar } from "notistack";
+import { LocalUserContext } from "./LocalUserContext";
+import { ArrowsClockwise } from "phosphor-react";
+import { SaveToFirestore, generateId } from "./Firestore";
+import * as Yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useForm } from "react-hook-form";
+import { createNewWatchlist } from "../Util/ListUtil";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { auth } from "./Firebase";
+
+export default function ProfileImportDialog({ subheadStyle }) {
+  const [open, setOpen] = useState(false);
+  const [user] = useAuthState(auth);
+
+  const [localUser, setLocalUser] = useContext(LocalUserContext);
+
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+
+  const [loading, setLoading] = useState(false);
+
+  const [accountName, setAccountName] = useState("");
+
+  const handleOpen = () => setOpen(true);
+
+  const handleClose = () => {
+    setOpen(false);
+    setAccountName("");
+    clearErrors();
+  };
+
+  function handleFormSubmission() {
+    clearErrors("accountName");
+    if (!errors.accountName) {
```

Comment:
instead of nesting, I often like a "validate, then act" structure for functions.  if we fail validation, we can take an early out (or throw).  maybe something like:
```js
if (errors.accountName) {
  return; // or throw
}

if (loading) {
  return; // or throw
}

// do action...
```

[This video](https://www.youtube.com/watch?v=CFRhGnuXG-4) explains the philosophy nicely 😀

Reply:
Great comment. In hindsight, `handleFormSubmission` will only fire if the form submittal passes validation.  

So, we're able to eliminate all of the conditionals I originally had. 🎉🎉🎉